include/ast.hpp

```diff
@@ -351,6 +375,21 @@ struct ArgExprNode : public ExprNode {
   std::unique_ptr<ExprNode> arg;
 };
 
+// @brief An array subscripting expression.
+struct ArrSubExprNode : public ExprNode {
+  ArrSubExprNode(Location loc, std::unique_ptr<ExprNode> postfix_expr,
+                 std::unique_ptr<ExprNode> index)
+      : ExprNode{loc},
+        postfix_expr{std::move(postfix_expr)},
+        index{std::move(index)} {}
+
+  void Accept(NonModifyingVisitor&) const override;
+  void Accept(ModifyingVisitor&) override;
+
+  std::unique_ptr<ExprNode> postfix_expr;
```

Comment:
I think `expr` suffices in this case; the term `postfix` confuses me. :thinking: